docs/arcadia-users-group/20240305-github-templates/lesson.md

```diff
@@ -0,0 +1,71 @@
+- [.github/lint.yml](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.github/lint.yml): The `.github/lint.yml` file is a GitHub Actions configuration file that sets up linting for the repository. Linting automatically checks the code for stylistic errors, programming errors, and other potential issues according to predefined rules. This helps maintain code quality and consistency across the project, making it easier to read, maintain, and contribute to.
+- [.pre-commit-config.yaml](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.pre-commit-config.yaml): The `.pre-commit-config.yaml` file configures pre-commit hooks for the repository. Pre-commit hooks are scripts that run automatically on each commit to check for common issues, such as syntax errors, formatting issues, or failing tests. This setup helps catch and fix problems early in the development process, improving code quality and reducing the time spent on code reviews.
+- [.vscode/extensions.json](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.vscode/extensions.json): The `.vscode/extensions.json` file is a configuration file for Visual Studio Code that specifies recommended extensions for the project. This file will prompt VS Code to install the ruff linter.
+- [Makefile](https://github.com/Arcadia-Science/python-analysis-template/tree/main/Makefile): The `Makefile` helps automate code quality and style checks, making it easier for developers to run all of these checks at once.
+
+## Benefits of templates
+
+
+- **Low effort for best practices:** The python analysis template repository have 10 files. While none of these files are complicated on their own, creating each of them takes time and domain knowledge of tools and syntax. Further, if each user had to create each file on their own, there is a lot of room for error. Using a template saves this time and cuts down on the number of mistakes that users create in their repos while reinforcing best practices.
+- **Reviewed once:** When a repository is started from a template, all of the code and files from that template are instantiated on the `main` branch. As you make your changes, reviewers only need to review the things you change, not all of the files from the template. This saves reviewers time because they don't have to review similar or idential text between many different repositories -- the content of a README.md or GitHub action workflow are reviewed when they are added to or changed in the template only. Overall, this saves reviewers time.
```

Comment:
TIL that this is a benefit of templates! Haven't dealt with this myself but it makes sense.

Reply:
LOL mert made this clear to me by reviewing the same text over. and over. and over. in my repo readmes 😂